linera-chain/src/chain.rs

```diff
@@ -178,8 +177,9 @@ where
     /// have been properly received by now.
     pub async fn validate_incoming_messages(&mut self) -> Result<(), ChainError> {
         let chain_id = self.chain_id();
-        for origin in self.inboxes.indices().await? {
-            let inbox = self.inboxes.load_entry(&origin).await?;
+        let origins = self.inboxes.indices().await?;
+        let inboxes = self.inboxes.try_load_entries(origins.clone()).await?;
+        for (origin, inbox) in origins.into_iter().zip(inboxes) {
```

Comment:
This `zip` seems like a recurring pattern. I wonder if we should change the signature of `try_load_entries` so that it returns a vector of `(key, view)` ?